cli/src/main.rs

```diff
@@ -841,3 +983,48 @@ fn get_pair_from_str(account: &str) -> sr25519::AppPair {
 		},
 	}
 }
+
+pub fn count_exchange_rate_update_events<P: Pair, Client: 'static>(
+	api: Api<P, Client>,
+	duration: Duration,
+) -> i32
+where
+	MultiSignature: From<P::Signature>,
+	Client: RpcClient + Subscriber + Send,
+{
+	let stop = duration_now() + duration;
+
+	//subscribe to events
+	let (events_in, events_out) = channel();
+	api.subscribe_events(events_in).unwrap();
+	let mut count = 0;
+
+	while remaining_time(stop).unwrap_or_default() > Duration::ZERO {
+		let event_str = events_out.recv().unwrap();
+		let unhex = Vec::from_hex(event_str).unwrap();
+		let mut er_enc = unhex.as_slice();
+		let events = Vec::<frame_system::EventRecord<Event, Hash>>::decode(&mut er_enc);
+		if let Ok(evts) = events {
+			for evr in &evts {
```

Comment:
Please don't abbreviate variables names - use `event_record` and `event_records`. And the variable `events` above, is probably an `events_result` ?